src/widgetastic/widget/table.py

```diff
@@ -0,0 +1,1055 @@
+        except AttributeError:
+            raise AttributeError('no {} attribute in class {}'.format(attr, type(self.refers_to)))
+
+    def __repr__(self):
+        return '{}({!r})'.format(type(self).__name__, self.refers_to)
+
+
+class TableRow(Widget, ClickableMixin):
+    """Represents a row in the table.
+
+    If subclassing and also changing the Column class, do not forget to set the Column to the new
+    class.
+
+    Args:
+        index: Position of the row in the table.
+    """
+    Column = TableColumn
+
+    def __init__(self, parent, index, logger=None):
+        Widget.__init__(self, parent, logger=logger)
+        # todo: get rid of all these indexes
+        needs_increment = parent._is_header_in_body or self.table.has_rowcolspan
+        self.index = index + 1 if needs_increment else index or 1
+
+    @property
+    def table(self):
+        return self.parent
+
+    def __repr__(self):
+        return '{}({!r}, {!r})'.format(type(self).__name__, self.parent, self.index)
+
+    def __locator__(self):
+        loc = self.parent.ROW_AT_INDEX.format(self.index)
+        return self.browser.element(loc, parent=self.parent)
+
+    def position_to_column_name(self, position):
+        """Maps the position index into the column name (pretty)"""
+        return self.table.index_header_mapping[position]
+
+    def __getitem__(self, item):
+        if isinstance(item, six.string_types):
+            index = self.table.header_index_mapping[self.table.ensure_normal(item)]
+        elif isinstance(item, int):
+            index = item
+        else:
+            raise TypeError('row[] accepts only integers and strings')
+
+        if self.table.table_tree:
+            # todo: add support of xpath and/or iteration to anytree lib
+            return self.table.resolver.glob(self.table.table_tree,
+                                            '/table/tbody/tr[{}]/*[{}]'.format(self.index-1,
+                                                                               index))[0].obj
+
+        else:
+            return self.Column(self, index, logger=create_item_logger(self.logger, item))
+
+    def __getattr__(self, attr):
+        try:
+            return self[self.table.ensure_normal(attr)]
+        except KeyError:
+            raise AttributeError('Cannot find column {} in the table'.format(attr))
+
+    def __dir__(self):
+        parent_dir = getattr(super(TableRow, self), '__dir__', None)
+        result = parent_dir if parent_dir else []
+        result.extend(self.table.attributized_headers.keys())
+        return sorted(result)
+
+    def __iter__(self):
+        for i, header in enumerate(self.table.headers):
+            yield header, self[i]
+
+    def read(self):
+        """Read the row - the result is a dictionary"""
+        result = {}
+        for i, (header, cell) in enumerate(self):
+            if header is None:
+                header = i
+            result[header] = cell.read()
```

Comment:
`result[header or i]` should allow you to remove the if statement and header=i assignment